pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -100,6 +100,68 @@ func (c *Controller) GetAppliedToGroupNum() int {
 	return c.ruleCache.GetAppliedToGroupNum()
 }
 
+// TODO: save namespace information in cache.
+func (c *Controller) GetNetworkPolicies() []v1beta1.NetworkPolicy {
+	var ret []v1beta1.NetworkPolicy
+	c.ruleCache.policySetLock.RLock()
```

Comment:
Just notice these functions are in Controller, which shouldn't access ruleCache's internal data and lock. Could you move these code to ruleCache and expose functions to Controller for the time being? I would expose the ruleCache to agent APIServer directly with another PR to avoid the unnecessary invoking proxy.

Reply:
Moved, I'm not sure is it follows your expectation.